header.go

```diff
@@ -509,16 +525,47 @@ func rfc2047AttributeName(s string) string {
 	if !strings.Contains(s, "?=") {
 		return s
 	}
-	pair := strings.SplitAfter(s, "?=")
-	// lets assume that the attribute was encoded because of unicode characters being present
-	// then the attribute value should be quoted
-	keyValuePair := strings.SplitAfter(decodeHeader(pair[0]), "=")
-	// only quote the parameter value if it isn't already quoted
-	if len(keyValuePair) > 1 {
-		if !strings.HasPrefix(keyValuePair[1], "\"") {
-			keyValuePair[1] = fmt.Sprintf("\"%s", keyValuePair[1])
+	// copy the string so we can return the original if we encounter any issues
+	temp := s
```

Comment:
`temp` is a code smell.  I think what I'd do here is replace`s` with `name` entirely in this function, and rename `temp` to `s`.  What do you think?

Reply:
I like it. Done.